framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1387,148 +1422,159 @@ private AnnotationMirror getAnnotation(
-            for (int i = 0; i < aiexprvals.size(); ++i) {
-                anaiexpr = aiexprvals.get(i);
-                if (anaiexpr instanceof FieldAccessExpr || anaiexpr instanceof NameExpr) {
-
-                    if (anaiexpr instanceof FieldAccessExpr) {
-                        elemarr[i] = findVariableElement((FieldAccessExpr) anaiexpr);
-                    } else {
-                        elemarr[i] = findVariableElement((NameExpr) anaiexpr);
-                    }
-
-                    if (elemarr[i] == null) {
-                        // A warning was already issued by findVariableElement;
-                        return;
-                    }
-                    String constval = (String) ((VariableElement) elemarr[i]).getConstantValue();
-                    if (constval != null) {
-                        elemarr[i] = constval;
-                    }
-                } else if (anaiexpr instanceof IntegerLiteralExpr) {
-                    elemarr[i] = Integer.valueOf(((IntegerLiteralExpr) anaiexpr).getValue());
-                } else if (anaiexpr instanceof StringLiteralExpr) {
-                    elemarr[i] = ((StringLiteralExpr) anaiexpr).getValue();
-                } else {
-                    ErrorReporter.errorAbort(
-                            "StubParser: unhandled annotation attribute type: " + anaiexpr);
-                }
+            for (int i = 0; i < arrayExpressions.size(); ++i) {
+                values[i] =
+                        getValueOExpressionInAnnotation(name, arrayExpressions.get(i), valueKind);
             }
-
-            builder.setValue(name, elemarr);
-        } else if (expr instanceof BooleanLiteralExpr) {
-            BooleanLiteralExpr blexpr = (BooleanLiteralExpr) expr;
-            ExecutableElement var = builder.findElement(name);
-            TypeMirror expected = var.getReturnType();
-            if (expected.getKind() == TypeKind.BOOLEAN) {
-                builder.setValue(name, blexpr.getValue());
-            } else if (expected.getKind() == TypeKind.ARRAY) {
-                Boolean[] arr = {blexpr.getValue()};
-                builder.setValue(name, arr);
+            builder.setValue(name, values);
+        } else {
+            Object value = getValueOExpressionInAnnotation(name, expr, valueKind);
+            if (expected.getKind() == TypeKind.ARRAY) {
+                Object[] valueArray = {value};
+                builder.setValue(name, valueArray);
             } else {
-                ErrorReporter.errorAbort(
-                        "StubParser: unhandled annotation attribute type: "
-                                + blexpr
-                                + " and expected: "
-                                + expected);
+                builderSetValue(builder, name, value);
             }
+        }
+    }
+
+    /**
+     * Cast to non-array values so that correct the correct AnnotationBuilder#setValue method is
+     * called.
```

Comment:
Can you add a comment why this is preferable to making the `AnnotationBuilder#setValue(CharSequence key, Object value)` method visible? I guess it is to be less error-prone with method overloading. Maybe making this a helper method somewhere closer to AnnotationBuilder would be useful?